ProtectionScan/Features/MainFeature.cs

```diff
@@ -199,5 +211,59 @@ private void WriteProtectionResultFile(string path, Dictionary<string, List<stri
             // Dispose of the writer
             sw?.Dispose();
         }
+        
+#if NETCOREAPP
+        /// <summary>
+        /// Write the protection results from a single path to a json file, if possible
+        /// </summary>
+        /// <param name="path">File or directory path</param>
+        /// <param name="protections">Dictionary of protections found, if any</param>
+        private static void WriteProtectionResultJson(string path, Dictionary<string, List<string>> protections)
+        {
+            if (protections == null)
+            {
+                Console.WriteLine($"No protections found for {path}");
+                return;
+            }
+
+            // Attempt to open a protection file for writing
+            StreamWriter? jsw = null;
+            try
+            {
+                jsw = new StreamWriter(File.OpenWrite($"protection-{DateTime.Now:yyyy-MM-dd_HHmmss.ffff}.json"));
+            }
+            catch
+            {
+                Console.WriteLine("Could not open protection log file for writing. Only a console log will be provided.");
+            }
+            
+            // Create the output data
+            string serializedData = System.Text.Json.JsonSerializer.Serialize(protections, JsonSerializerOptions);
+
+            // Write the output data
+            // TODO: this prints plus symbols wrong, probably some other things too
+            jsw?.WriteLine(serializedData);
+            jsw?.Flush();
+            
+            // Dispose of the writer
+            jsw?.Dispose();
+        }
+        
+        /// <summary>
+        /// JSON serializer options for output printing
+        /// </summary>
+        private static System.Text.Json.JsonSerializerOptions JsonSerializerOptions
+        {
+            get
+            {
+#if NETCOREAPP3_1
+                var serializer = new System.Text.Json.JsonSerializerOptions { WriteIndented = true };
+#else
+                var serializer = new System.Text.Json.JsonSerializerOptions { IncludeFields = true, WriteIndented = true };
+#endif
```

Comment:
This extra block is unnecessary. Since you're outputting dictionary data, there are no fields to worry about.